packages/tasks_panel/src/view/tasks-tree.ts

```diff
@@ -23,26 +33,59 @@ export class TasksTree implements TreeDataProvider<TreeItem> {
     this._onDidChangeTreeData.fire(null);
   }
 
-  public async getChildren(element?: TreeItem): Promise<TreeItem[]> {
-    const tasks = await this.tasksProvider.getConfiguredTasks();
-    if (element === undefined) {
-      const intents = uniq(map(tasks, (_) => _.__intent));
-      getClassLogger(LOGGER_CLASS_NAME).debug(messages.GET_TREE_INTENTS(intents.length));
-      return map(intents, (_) => new IntentTreeItem(_, TreeItemCollapsibleState.Collapsed));
-    }
-    const intent = element.label ?? "";
-    const tasksByIntent = filter(tasks, (_) => _.__intent === intent);
-    const result = map(
-      tasksByIntent,
+  private filterByFolder(tasks: ConfiguredTask[], parent?: ProjectTreeItem): ConfiguredTask[] {
+    return parent ? filter(tasks, ["__wsFolder", parent.fqn]) : tasks;
+  }
+
+  private getIntents(tasks: ConfiguredTask[], parent?: ProjectTreeItem): IntentTreeItem[] {
+    const intents = sortBy(uniq(map(this.filterByFolder(tasks, parent), "__intent")));
+    getClassLogger(LOGGER_CLASS_NAME).debug(messages.GET_TREE_BRANCHES("intent", intents.length));
+    return map(intents, (_) => new IntentTreeItem(_, TreeItemCollapsibleState.Collapsed, parent));
+  }
+
+  private getWorkspaces(wsFolders: string[]): IntentTreeItem[] {
+    getClassLogger(LOGGER_CLASS_NAME).debug(messages.GET_TREE_BRANCHES("workspace", wsFolders.length));
+    return map(
+      wsFolders,
+      (_) =>
```

Comment:
why you are using "_" and not a variable name (e.g. wsFolderPath)?

Reply:
changed